src/app.js

```diff
@@ -1,61 +1,99 @@
 import './style.css';
-import { setCompleted } from './function.js';
+import {
+  createlocalstorage, addBtn, listArr,
+} from './script.js';
 
 const container = document.querySelector('.list-container');
+const deleteallbtn = document.querySelector('.clear');
 
-let todoList = [
-  {
-    index: 0,
-    description: 'Wash the dishes',
-    completed: true,
-  },
-
-  {
-    index: 1,
-    description: 'Complete todo lis project',
-    completed: true,
-  },
-
-  {
-    index: 2,
-    description: 'Play game with my girlfriend',
-    completed: true,
-  },
-
-  {
-    index: 3,
-    description: 'Eat some cookies',
-    completed: true,
-  },
-];
-
-const storedData = JSON.parse(localStorage.getItem('todos'));
-if (storedData) {
-  todoList = storedData;
-}
-
-const display = () => {
-  todoList.forEach((item) => {
-    const containerItems = `
-    <div class="task">
-    <div class="wrapper">
-        <input type="checkbox" name="" id="boxes" class="box m-input" data-id=${item.index} ${item.completed ? 'checked' : ''}>
-        <input type="text" name="" data-id="${item.index}" value="${item.description}">
-    </div>
-        <i id="ellipse" class="fas fa-ellipsis-v m-i"></i>
-        
-    </div>
-    <hr>`;
-    container.innerHTML += containerItems;
+const display = (array) => {
+  let containerItems = '';
+  array.forEach((item) => {
+    containerItems += `
+      <div class="task">
+      <div class="wrapper">
+          <input type="checkbox" name="" id="boxes" class="box m-input" data-id=${item.index} ${item.completed ? 'checked' : ''}>
+          <input type="text" name="" data-id="${item.index}" id="${item.index}" class="input-text" value="${item.description}">
+      </div>
+          <i id="ellipse" class="fas fa-ellipsis-v m-i"></i>
+         <i id="deletebtn" class="fas fa-trash"></i>
+          
+      </div>
+      <hr>`;
   });
 
+  container.innerHTML += containerItems;
+
   const boxes = document.querySelectorAll('.box');
 
   boxes.forEach((box) => {
     box.addEventListener('change', () => {
-      setCompleted(todoList, box);
+      const inputtext = document.getElementById(box.dataset.id);
+
+      const founditem = listArr.find((item) => box.dataset.id === item.index.toString());
+      founditem.completed = box.checked;
+      if (founditem.completed === true) {
+        inputtext.style.textDecoration = 'line-through';
+      } else {
+        inputtext.style.textDecoration = 'none';
+      }
+      localStorage.setItem('todos', JSON.stringify(listArr));
     });
   });
 };
 
-display();
+addBtn.addEventListener('click', () => {
+  createlocalstorage(listArr);
+  window.location.reload();
+});
+display(listArr);
+
+const task = document.querySelectorAll('.task');
+
+function deleteTask(idx) {
+  const storage = JSON.parse(localStorage.getItem('todos'));
+  const savedata = storage.filter((value, index) => index !== idx);
+  localStorage.setItem('todos', JSON.stringify(savedata));
+  window.location.reload();
+}
```

Comment:
- [ ] When a task is deleated the indexes should be reasigned, for example if i have 2 task with index 1 and 2 and i delete the first task, the task that remains should now have index 1.
    - As you can see in the first screenshot that is not happening.
    
     ![imagen](https://user-images.githubusercontent.com/60789317/136599915-4bfc02b0-abee-4e42-bdc8-7a17a5773b66.png)

     ![imagen](https://user-images.githubusercontent.com/60789317/136599689-c4b6dbd4-f1bb-43ab-8c98-e83b37c88684.png)